### PR TITLE
Fix decoder when processing template objects in newapp

### PR DESCRIPTION
**Pull request by soltysh**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1671528
/assign @juanvallejo

### CODE REVIEW

**Comment by soltysh** [comment]
/hold 
there's a problem in test-cmd I'm looking at right now

**Comment by soltysh** [comment]
/hold
@deads2k does this look legit?

File: pkg/oc/lib/newapp/cmd/newapp.go [hunk 1]
```diff
@@ -170,6 +187,31 @@ func (e ErrRequiresExplicitAccess) Error() string {
 // ErrNoInputs is returned when no inputs are specified
 var ErrNoInputs = errors.New("no inputs provided")
 
+// TODO: we should run entire new-app using unstructured only
```
Comment (soltysh) [inline pkg/oc/lib/newapp/cmd/newapp.go]: @deads2k that's the nasty I've mentioned you yesterday, but I'm leaving a TODO here, when we have time we need to switch entire new-app to work unstructured only -> https://github.com/openshift/origin/issues/21983

**Comment by deads2k** [comment]
This makes it clear that we are one type away from being broken again.

make it green and we can have it.

**Comment by soltysh** [comment]
> This makes it clear that we are one type away from being broken again.

That's why I'll start crafting a PR switching entire new-app to unstructured, we may not land it for 4.0, but it'll be ready when the gates open for 4.1. 

/retest

**Comment by soltysh** [comment]
Based on approval self-tagging.

**Comment by soltysh** [comment]
/hold 
we've decided not to do it this way. I'll create a jira card to track this for the devexp team.
/close

**Pull request closed without merging**